src/lib/comms/include/sol-lwm2m.h

```diff
@@ -394,6 +394,28 @@ struct sol_lwm2m_resource {
         (ret_value_) = sol_lwm2m_resource_init((resource_), (id_), 1, SOL_LWM2M_RESOURCE_DATA_TYPE_INT, SOL_TYPE_CHECK(int64_t, (value_))); \
     } while (0)
 
+
+/**
+ * @brief A payload received from the network used to create a LWM2M object instance.
+ *
+ * @see sol_lwm2m_content_type
+ * @see struct sol_lwm2m_object::create
+ */
+struct sol_lwm2m_payload {
+    /** @brief The payload type
+     * @see sol_lwm2m_content_type
+     */
+    enum sol_lwm2m_content_type type;
+    union sol_lwm2m_payload_data {
+        /** @brief The payload content as TLV format -
+            use only when @c sol_lwm2m_payload::type is #SOL_LWM2M_CONTENT_TYPE_TLV
```

Comment:
This line is missing the initial `*`. I don't know if it's mandatory for doxygen though.